src/main/java/com/rybka/service/ExchangeService.java

```diff
@@ -1,100 +1,73 @@
-
-            currencyValue = jsonObject.get(userTargetCurrencyAbbreviation).getAsDouble();
-            baseCurrencyAbbreviation = userBaseCurrencyAbbreviation;
-            targetCurrencyAbbreviation = userTargetCurrencyAbbreviation;
-        } catch (Exception e) {
-            System.out.println("Incorrect exchange data!");
+    public Currency loadCurrencyOf(String userBaseCurrencyAbbreviation, String userTargetCurrencyAbbreviation,
+                                   Double amount) {
+        var response = sendRequest(userBaseCurrencyAbbreviation);
+        var currencyRate = retrieveJsonData(response, userTargetCurrencyAbbreviation);
+        if (currencyRate == 0.0) {
+            System.out.println("Invalid currency abbreviation");
             System.exit(3);
         }
+        return constructDataObject(currencyRate, userBaseCurrencyAbbreviation, userTargetCurrencyAbbreviation, amount);
     }
 
-    public void exchange(Double amount) {
-
-        this.amount = amount;
-        this.total = amount * currencyValue;
-    }
-
-    public void getCurrencyObject() {
-        Currency currency = new Currency(baseCurrencyAbbreviation, targetCurrencyAbbreviation, currencyValue, amount, total);
-        HibernateDAO hibernateDAO = new HibernateDAO();
-        hibernateDAO.save(currency);
-        hibernateDAO.showTableRow();
-    }
-
-    public Double getCurrencyValue() {
-        return currencyValue;
-    }
-
-    public void setCurrencyValue(Double currencyValue) {
-        this.currencyValue = currencyValue;
-    }
-
-    public String getBaseCurrencyAbbreviation() {
-        return baseCurrencyAbbreviation;
+    private Currency constructDataObject(Double currencyRate, String userBaseCurrencyAbbreviation,
+                                         String userTargetCurrencyAbbreviation, Double amount) {
+        currency.setCurrencyValue(currencyRate);
+        currency.setBaseCurrencyAbbreviation(userBaseCurrencyAbbreviation);
+        currency.setTargetCurrencyAbbreviation(userTargetCurrencyAbbreviation);
+        currency.setAmount(amount);
+        currency.setTotal(currency.getAmount() * currency.getCurrencyValue());
+        return currency;
     }
 
-    public void setBaseCurrencyAbbreviation(String baseCurrencyAbbreviation) {
-        this.baseCurrencyAbbreviation = baseCurrencyAbbreviation;
-    }
-
-    public String getTargetCurrencyAbbreviation() {
-        return targetCurrencyAbbreviation;
-    }
-
-    public void setTargetCurrencyAbbreviation(String targetCurrencyAbbreviation) {
-        this.targetCurrencyAbbreviation = targetCurrencyAbbreviation;
-    }
-
-    public Double getAmount() {
-        return amount;
-    }
-
-    public void setAmount(Double amount) {
-        this.amount = amount;
-    }
-
-    public Double getTotal() {
-        return total;
+    private HttpResponse<String> sendRequest(String userBaseCurrencyAbbreviation) {
```

Comment:
As we discussed we can move this method to the separate service (api connector). 
Also rename this method name :_

Reply:
fixed